lib/index.js

```diff
@@ -432,125 +432,137 @@ Utils.parseDateTimeField = function(field, options) {
+      // Otherwise, parse the string to create a date in the future.
+      parsedDate =
+        withoutOffsetModifiers === '' && hasOffsetModifier ?
+          Utils._getCurrentDate(options) :
+          Utils._getFutureDate(withoutOffsetModifiers, options);
+
+      if(SugarDate.isValid(parsedDate) && hasOffsetModifier && offsetSecs) {
+        // Apply the offset modifier.
+        // If it is negative, it will subtract.
+        SugarDate.addSeconds(parsedDate, offsetSecs);
+      }
+    }
 
-    // Run the regex on the string to strip out any
-    // offset modifiers, adding/subtracting
-    // them from the overall offsetSecs.
-    var offsetSecs = 0;
-    var hasOffsetModifier = false;
-
-    var withoutOffsetModifiers = field
-      .replace(
-        offsetRegex,
-        function(match, p1, p2, p3) {
-          if(p1 === '+') {
-            // We should increment the offsetSecs
-            // by the appropriate amount.
-            offsetSecs = offsetSecs + (offsetMultipliers[p3] * p2);
-
-          } else {
-            // We should decrement the offsetSecs
-            // by the appropriate amount.
-            offsetSecs = offsetSecs - (offsetMultipliers[p3] * p2);
-          }
-
-          // Mark the fact we have at least one offset modifier.
-          hasOffsetModifier = true;
-
-          // Return a blank string, to remove the
-          // offset modifier from the original string.
-          // This is so that the offset modifier does
-          // not interfere with the datetime parsing.
-          return '';
-        })
-      .trim();
-
-    // If we only have offset modifiers, initialise the date as now.
-    // Otherwise, parse the string to create a date in the future.
-    parsedDate =
-      withoutOffsetModifiers === '' && hasOffsetModifier ?
-        Utils._getCurrentDate(options) :
-        Utils._getFutureDate(withoutOffsetModifiers, options);
-
-    if(SugarDate.isValid(parsedDate) && hasOffsetModifier && offsetSecs) {
-      // Apply the offset modifier.
-      // If it is negative, it will subtract.
-      SugarDate.addSeconds(parsedDate, offsetSecs);
+    if (previousNewDateInternal) {
+      Utils.Logger.debug('Utils.parseDateTimeField: parseDateTime: Tearing down newDateInternal');
+      SugarDate.setOption('newDateInternal', previousNewDateInternal);
     }
-  }
 
-  if (previousNewDateInternal) {
-    SugarDate.setOption('newDateInternal', previousNewDateInternal);
+    return parsedDate;
   }
 
-  var dateMoment;
-  if (SugarDate.isValid(parsedDate)) {
+  var parsedDate = parseDateTime(field, options);
+  var rtnObject = getRtnObject(parsedDate, SugarDate.isValid(parsedDate));
+
+  var parsedDateWithTimezone = parseDateTime(field, options, timezone);
+  if (SugarDate.isValid(parsedDateWithTimezone)) {
+    Utils.Logger.debug('Utils.parseDateTimeField: parsedDateWithTimezone:', parsedDateWithTimezone.toISOString());
+
     var hasTZOffset = new RegExp('[+-]{1}[0-9]{2}:?[0-9]{2}').test(field);
```

Comment:
I know this wasn't part of the changes but maybe this regex could be tightened up? How about:

```
/(?:[zZ]|[+-][01]\d:[0-5]\d)$/.test(field)
```

This takes into account:

- a UTC `Z` timezone offset
- the fact that `+99:99` isn't a valid timezone offset.